contracts/modules/cover/Cover.sol

```diff
@@ -346,43 +438,45 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard, Mu
 
         // check if this request should be skipped, keeping the previous allocation
         if (poolAllocationRequests[i].skip) {
-          coverSegmentAllocations[allocationRequest.coverId][segmentId].push(previousPoolAllocation);
-          totalCoverAmountInNXM += previousPoolAllocation.coverAmountInNXM;
+          coverSegmentAllocations[allocationRequest.coverId][params.segmentId].push(previousPoolAllocation);
+          vars.totalCoverAmountInNXM += previousPoolAllocation.coverAmountInNXM;
           continue;
```

Comment:
Two important observations regarding this bit of logic:
1. You always count towards `totalCoverAmountInNXM` the cover amount from this allocation, although the pool is skipped. This works fine when the user only performs an increase of the amount. However, if the user (also) increases the period, skipping a pool has a different semantic, and it's rather equivalent to fully abandoning the pool (give up the premium already paid on it and don't count that allocation towards the total cover amount), as that pool probably doesn't have the required capacity for the newly desired period. What should the behaviour be in this case?
2. As you pointed out @danoctavian, for skipped pools we never reprice the allocations, as we do for non-skipped pools. Is this a fair behaviour?

Reply:
The spec says something else for skipped pools:
```
- For Pools that we “skip”:
    - deallocate the capacity, but rewards keep streaming to the stakers as per the original allocation
    - set `coverAmountInNXM` to 0```

Reply:
At this point given that one can simply leave a pool in by submitting the same amount for that allocation and set request.period == 0 (no extra period), the skip logic doesn't seem necessary.

With that existing implementation of the skip boolean one had to check that only the amount is increased and not the period to apply it.